### ISSUE TITLE
api key authorization configuration thru swaggerUiAuthProvider.credentials not working

### ISSUE THREAD

**Issue by panlina** [Orange-OpenSource/angular-swagger-ui#93]
I configured api key as in the example:
```
.config(function (swaggerUiAuthProvider) {
	swaggerUiAuthProvider.credentials({
		apiKey: {
			Authorization: "abc"
		}
	});
})
```
my swagger.json looks like:
```
"securityDefinitions": {
    "jwt": {
        "type": "apiKey",
        "name": "Authorization",
        "in": "header"
    }
},
"security": [
    {
        "jwt": []
    }
],
```
However, `swaggerUiAuthProvider.credentials` does not seem to work. There's no `Authorization` header in the request if I don't use the authorization button to authorize manually. When the authorization modal pops up, the value I configured in the code is not populated either. It just looks like the configuration does not take effect at all.
Did I miss something?

**Comment by mathieuales** [comment]
Hi !
The right syntax is:

```
.config(function (swaggerUiAuthProvider) {
	swaggerUiAuthProvider.credentials({
		apiKey: {
			jwt: "abc"
		}
	});
})

```

**Comment by panlina** [comment]
Tried it and the value is now populated in the authorization modal. Cool, but it still requires manually open the modal to authorize to take effect. I suppose it to work as if it is already authorized. Is it by design?
To clarify, what my "authorized" really means is "authorization info is set" since the actual authorization take place after the request is sent. So I mean I suppose setting the apikey in the code is equivalent to setting it in the modal so that I do not need to authorize again.

**Comment by mathieuales** [comment]
This is how it works for now.
If you want authorization being auto validated when you set apiKey in code you have to write your own module this way:

```js
angular
	.service('yourModule', function($q) {

                this.execute = function(data) {
                    var deferred = $q.defer();
                    angular.forEach(data.openApiSpec.securityDefinitions, function(securityDefinition, name) {
                        if (name = 'jwt') {
                            securityDefinition.apiKey = 'abc';
                            securityDefinition.valid = true;
                        }
                    });
                    deferred.resolve(true);
                    return deferred.promise;
                }

            })
            .run(function(swaggerModules, yourModule){
                swaggerModules.add(swaggerModules.BEFORE_DISPLAY, yourModule, 1);
            })
	...

```

**Comment by panlina** [comment]
Seems to work. Thanks!

**Comment by mathieuales** [comment]
Hi!
Auto-validation included since 0.6.0 version
Please upgrade and update configuration as describe in readme
